### PR TITLE
HBASE-22820 Do not need to persist default rs group now

### PR DESCRIPTION
**Pull request by sunhelly**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 122 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ HBASE-22514 Compile Tests _ |
| +1 | mvninstall | 518 | HBASE-22514 passed |
| +1 | compile | 83 | HBASE-22514 passed |
| +1 | checkstyle | 100 | HBASE-22514 passed |
| +1 | shadedjars | 369 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 49 | HBASE-22514 passed |
| 0 | spotbugs | 328 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 326 | HBASE-22514 passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 408 | the patch passed |
| +1 | compile | 74 | the patch passed |
| +1 | javac | 74 | the patch passed |
| +1 | checkstyle | 98 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 359 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 1257 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 38 | the patch passed |
| +1 | findbugs | 297 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 13882 | hbase-server in the patch failed. |
| +1 | asflicense | 30 | The patch does not generate ASF License warnings. |
| | | 18488 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.replication.regionserver.TestRegionReplicaReplicationEndpoint |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.0 Server=19.03.0 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-482/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/482 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 7f8c75660dcf 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-482/out/precommit/personality/provided.sh |
| git revision | HBASE-22514 / 595aa9df38 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-482/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-482/1/testReport/ |
| Max. process+thread count | 4967 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-482/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by sunhelly** [comment]
Failed uts are not relevant to the changes.
Ping @Apache9 @infraio for reviewing.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupInfoManagerImpl.java [hunk 1]
```diff
@@ -456,7 +458,12 @@ private synchronized void flushConfig(Map<String, RSGroupInfo> newGroupMap) thro
 
     // Make changes visible after having been persisted to the source of truth
     resetRSGroupMap(newGroupMap);
+    saveRSGroupMapToZK(newGroupMap);
 
+    updateCacheOfRSGroups(newGroupMap.keySet());
+  }
+
+  public void saveRSGroupMapToZK(Map<String, RSGroupInfo> newGroupMap) throws IOException {
```
Comment (Apache9) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupInfoManagerImpl.java]: Does this method need to be public?

Comment (sunhelly) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupInfoManagerImpl.java]: Changed to private.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 73 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ HBASE-22514 Compile Tests _ |
| +1 | mvninstall | 355 | HBASE-22514 passed |
| +1 | compile | 59 | HBASE-22514 passed |
| +1 | checkstyle | 92 | HBASE-22514 passed |
| +1 | shadedjars | 305 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 37 | HBASE-22514 passed |
| 0 | spotbugs | 276 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 275 | HBASE-22514 passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 331 | the patch passed |
| +1 | compile | 56 | the patch passed |
| +1 | javac | 56 | the patch passed |
| +1 | checkstyle | 87 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 305 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 1025 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 37 | the patch passed |
| +1 | findbugs | 278 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 15104 | hbase-server in the patch failed. |
| +1 | asflicense | 33 | The patch does not generate ASF License warnings. |
| | | 18861 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.regionserver.TestSplitTransactionOnCluster |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-482/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/482 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux b4998e584579 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-482/out/precommit/personality/provided.sh |
| git revision | HBASE-22514 / 07fe41db42 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-482/4/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-482/4/testReport/ |
| Max. process+thread count | 4258 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-482/4/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request merged**